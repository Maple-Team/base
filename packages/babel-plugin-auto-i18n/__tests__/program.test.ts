import { getTransformCode } from './test-help'

describe('detect has import react-i18n', () => {
  it('case 1', () => {
    const sourceCode = `
    import {useTranslation} from 'react-i18next'
    export const Component = () => {
      return <div name="属性">站位文本</div>
    }
    `
    const result = getTransformCode(sourceCode, 'imported-i18n1.tsx')
    expect(result?.code).toMatchSnapshot()
  })
  it('case 2', () => {
    const sourceCode = `
    export const Component = () => {
      return <div name="属性">站位文本</div>
    }
    `
    const result = getTransformCode(sourceCode, 'imported-i18n2.tsx')
    expect(result?.code).toMatchSnapshot()
  })
})
