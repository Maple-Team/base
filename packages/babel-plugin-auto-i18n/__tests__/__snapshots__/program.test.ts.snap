// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detect has import react-i18n case 1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
};"
`;

exports[`detect has import react-i18n case 2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div name={t(\\"key_属性\\")}>{t(\\"key_站位文本\\")}</div>;
};"
`;
